### DISCUSSION TITLE
Is there a specific format/function call we need to make to use the translate/rotate function.

### DISCUSSION THREAD

**Discussion by pyenthu** [elalish/manifold#879] [Q&A]
Just checking if there is another convention for calling the rotate/translate functions.
Spent a couple of hours but seem to be missing something really simple.
The composite works but I am unable to translate or rotate the manifold.
```
export function comp() {
  let mf = Manifold.cylinder(200, 20, 2, 30, true);
  for (let pp of tunnels) {
    mf = mf.add(Manifold.cylinder(pp.bot - pp.top, pp.dia));
  }
  let rot: Vec3 = [0, 90, 0];
  mf.rotate(rot);
  mf.translate([10, 300, 400]);
  let geom = mesh2geometry(mf.getMesh());
  geom.computeVertexNormals();
  return geom;
}
```

**Comment by pyenthu** [comment]
Sorry about. This. Should have perserved a bit more.
basically there is no in-situ action. It returns a new manifold which needs to be assigned.
this will work
`mf=mf.rotate([90,0,0]`
This will not
`mf.rotate([90,0,0])`

I hope it saves someone the time when starting.

**Reply by elalish** [reply]
Yeah, this is why we need to add `nodiscard` across our API - then the compiler will tell you this.
